File: pkg/server/delete-visualization.go
package server

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/sirupsen/logrus"

	"github.com/rbren/vizzy/pkg/files"
)

func deleteVisualization(c *gin.Context) {
	projectID := c.Param("projectID")
	if projectID == "" {
		c.JSON(http.StatusBadRequest, gin.H{"error": "Project ID is required"})
		return
	}
	visualizationID := c.Param("visualizationID")
	if visualizationID == "" {
		c.JSON(http.StatusBadRequest, gin.H{"error": "Visualization ID is required"})
		return
	}

	s3 := files.GetFileManager()
	vizKey := "projects/" + projectID + "/visualizations/" + visualizationID + "/"
	err := s3.DeleteRecursive(vizKey)
	if err != nil {
		logrus.WithError(err).Error("Failed to delete visualization")
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, gin.H{"success": true})
}

File: pkg/server/delete-visualization.py
from flask import Flask, request, jsonify
import logging

# Assuming this module is implemented similarly to its Go counterpart
from files import get_file_manager

app = Flask(__name__)
logging.basicConfig(level=logging.ERROR)

@app.route('/project/<project_id>/visualization/<visualization_id>', methods=['DELETE'])
def delete_visualization(project_id, visualization_id):
    if not project_id:
        return jsonify({"error": "Project ID is required"}), 400

    if not visualization_id:
        return jsonify({"error": "Visualization ID is required"}), 400

    s3 = get_file_manager()
    viz_key = f"projects/{project_id}/visualizations/{visualization_id}/"

    try:
        s3.delete_recursive(viz_key)
    except Exception as e:
        logging.error(f"Failed to delete visualization: {str(e)}")
        return jsonify({"error": str(e)}), 500

    return jsonify({"success": True}), 200

if __name__ == '__main__':
    app.run(debug=True)